src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -279,5 +304,61 @@ public boolean delete() {
         // TODO: Add journey state deletion.
         return Persistence.monitoredTrips.removeById(this.id);
     }
+
+    public Map<String, String> parseQueryParams() throws URISyntaxException {
+        return URLEncodedUtils.parse(
+            new URI(String.format("http://example.com/plan?%s", queryParams)),
+            UTF_8
+        ).stream().collect(Collectors.toMap(NameValuePair::getName, NameValuePair::getValue));
+    }
+
+    public boolean isArriveBy() throws URISyntaxException {
+        // if arriveBy is not included in query params, OTP will default to false, so initialize to false
+        return parseQueryParams().getOrDefault("arriveBy", "false").equals("true");
+    }
+
+    /**
+     * Gets the timezone of the target location. When planning a trip, we either want to depart at a certain time or 
+     * arrive by a certain time. When departing, we use the local time present at the origin and when arriving by we 
+     * use the local time at the destination. Therefore, this method will return the timezone at the destination if this
+     * trip is an arriveBy trip, or the timezone at the origin if the trip is a depart at trip.
+     */
+    public ZoneId timezoneForTargetLocation() throws URISyntaxException {
+        double lat, lon;
+        if (isArriveBy()) {
+            lat = to.lat;
+            lon = to.lon;
+        } else {
+            lat = from.lat;
+            lon = from.lon;
+        }
+        Optional<ZoneId> fromZoneId = getZoneIdForCoordinates(lat, lon);
+        if (fromZoneId.isEmpty()) {
+            String message = String.format(
+                "Could not find coordinate's (lat=%.6f, lon=%.6f) timezone for monitored trip %s",
+                lat,
+                lon,
+                id
+            );
+            throw new RuntimeException(message);
+        } 
+        return fromZoneId.get();
+    }
+
+    /**
+     * Returns the target hour of the day that the trip is either departing at or arriving by
+     */
+    @BsonIgnore
+    public int tripTimeHour() {
+        return Integer.valueOf(tripTime.split(":")[0]);
```

Comment:
There should still be a null check (at least) here and in `tripTimeMinutes`.

Reply:
The tripTime should be guaranteed to exist due to https://github.com/ibi-group/otp-middleware/pull/75#discussion_r493967601.